public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,823 @@
+    formikToDetectorName(embeddable.vis.title)
+  );
+  const [intervalValue, setIntervalalue] = useState(10);
+  const [delayValue, setDelayValue] = useState(1);
+  const [enabled, setEnabled] = useState<boolean>(false);
+  const [shingleSizeValue, setShingleSizeValue] = useState(8);
+
+  const title = embeddable.getTitle();
+  const onAccordionToggle = (key) => {
+    const newAccordionsOpen = { ...accordionsOpen };
+    newAccordionsOpen[key] = !accordionsOpen[key];
+    setAccordionsOpen(newAccordionsOpen);
+  };
+  const onDetectorNameChange = (e, field) => {
+    field.onChange(e);
+    setDetectorNameFromVis(e.target.value);
+  };
+  const onIntervalChange = (e, field) => {
+    field.onChange(e);
+    setIntervalalue(e.target.value);
+  };
+  const onDelayChange = (e, field) => {
+    field.onChange(e);
+    setDelayValue(e.target.value);
+  };
+  const aggList = embeddable.vis.data.aggs.aggs.filter(
+    (feature) => feature.schema == 'metric'
+  );
+  const featureList = aggList.filter(
+    (feature, index) => index < (aggList.length < 5 ? aggList.length : 5)
+  );
+
+  const getFeatureNameFromParams = (id) => {
+    let name = find(embeddable.vis.params.seriesParams, function (param) {
+      if (param.data.id === id) {
+        return true;
+      }
+    });
+
+    return name.data.label.replace(/[^a-zA-Z0-9-_]/g, '_');
+  };
+
+  const handleValidationAndSubmit = (formikProps) => {
+    if (!isEmpty(formikProps.errors)) {
+      focusOnFirstWrongFeature(formikProps.errors, formikProps.setFieldTouched);
+      notification.toasts.addDanger('One or more input fields is invalid');
+    } else {
+      handleSubmit(formikProps.values);
+    }
+  };
+
+  const handleSubmit = (values, formikHelpers) => {
+    try {
+      const detectorToCreate = formikValueToDetector(values);
+      dispatch(createDetector(detectorToCreate)).then(async (response) => {
+        dispatch(startDetector(response.response.id)).then(
+          (startDetectorResponse) => {
+            notification.toasts.addSuccess(`Detector created: ${values.name}`);
+          }
+        );
+        enum VisLayerTypes {
+          PointInTimeEvents = 'PointInTimeEvents',
+        }
+        const fn = {
+          type: VisLayerTypes.PointInTimeEvents,
+          name: 'overlay_anomalies',
+          args: {
+            detectorId: response.response.id,
+          },
+        } as VisLayerExpressionFn;
+
+        const savedObjectToCreate = {
+          title: embeddable.vis.title,
+          pluginResourceId: response.response.id,
+          visId: embeddable.vis.id,
+          savedObjectType: 'visualization',
+          visLayerExpressionFn: fn,
+        } as ISavedAugmentVis;
+
+        const savedObject = await createAugmentVisSavedObject(
+          savedObjectToCreate
+        );
+
+        const saveObjectResponse = await savedObject.save({});
+      });
+      closeFlyout();
+    } catch (e) {
+    } finally {
```

Comment:
We should propagate some error response here. There could be errors during the creation, during the start, or during the saving of the saved object. We have existing logic on the AD create page: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/ReviewAndCreate/containers/ReviewAndCreate.tsx#L235

Reply:
updated it with more error responses during the creation and the start. Not sure about the error wording for the saved object, will leave it to Amit @amitgalitz

Reply:
It should be consistent on both create & associate flows. Maybe you can work together to have a shared function (or at least copied logic) for both of these so they are consistent @jackiehanyang @amitgalitz

Reply:
this is discussed here: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/487#discussion_r1204688791, resolving this comment